Wikipedia/Code/TalkPageViewModel.swift

```diff
@@ -148,10 +146,13 @@ final class TalkPageViewModel {
             coffeeRollText = nil
         }
         
-        projectLanguage = languageCode
+        if let projectIconName = project.projectIconName {
+           projectSourceImage = UIImage(named: projectIconName)
```

Comment:
This is one where I'm unclear about our desired intent: a https://phabricator.wikimedia.org/T310273 screenshot seems to visually imply a fallback to the `.mediawiki` project icon in the user talk header on occurrences of `WikimediaProject.wikipedia` projects, but one of your comments there seems to imply we're ok with displaying no icons for `WikimediaProject.wikipedia`. I could be misinterpreting.

If the intent is indeed to be empty for `.wikipedia` as it is here, I'd recommend we do something to prevent the excess leading space present when no icon is set:

![leading_space](https://user-images.githubusercontent.com/5652327/195437248-f9b923c8-cb87-420f-9129-d6650a6c6a14.jpg)

Depending on if the `configure(...)` here may be called more than once or not, maybe something here https://github.com/wikimedia/wikipedia-ios/blob/b545a207482806cff031c31c2452faafb0b60310/Wikipedia/Code/TalkPageHeaderView.swift#L300

as simple as:

```swift
if let projectSourceImage = viewModel.projectSourceImage {
    projectImageView.image = projectSourceImage
} else {
    projectImageView.removeFromSuperview()
}
```


If you'd rather we address this in a separate PR though, no worries just lemme know!

Reply:
Excellent catch! I actually just assumed any talk page from the Wikipedia project would only show the language, because that's how we treated the Notifications Center cells. But that may not be the case.

I left a comment [here](https://phabricator.wikimedia.org/T310273#8313012). I'll address this bit of feedback when we have more clarity on the requirements.

Reply:
The extra space should now be fixed with your suggestion. Thanks for helping solidify [these requirements](https://phabricator.wikimedia.org/T310273#8313095).

Sidenote: There's an [exception](https://phabricator.wikimedia.org/T310273#8314520) with MediaWiki in particular (which is the icon from the mocks), which uses Flow and thus does NOT produce notifications links that the app recognizes as native talk page links. I checked with the backend team and the discussion tools API doesn't support Flow either. So we may never reach a point where the MediaWiki project icon displays in this header view, but I think it should be fine to keep the logic generic anyway. Should MediaWiki ever disable Flow and produce notifications with recognized talk page links, the app should start picking it up as a native talk page and display it appropriately.